tests/UI/pages/FO/checkout/index.js

```diff
@@ -113,6 +115,16 @@ class Checkout extends FOBasePage {
     return this.elementVisible(page, this.deliveryOptionLabel(shippingMethod), 2000);
   }
 
+  /**
+   * Is confirm button visible and enabled
+   * @param page
+   * @returns {Promise<boolean>}
+   */
+  isPaymentConfirmationButtonVisibleAndEnabled(page) {
+    // small side effect note, the selector is the one that checks for disabled
```

Comment:
> small side effect note, the selector is the one that checks for disabled

So no need to add selector `this.noPaymentNeededElement` ?

Reply:
`this.noPaymentNeededElement` is used in the test case.

What I meant with my comment is that the `this.paymentConfirmationButton` has the `[disabled]` selector. This might be confusing as the one might want to check that the button is there and what its state is. Also it looks like the disabled attribute is not even set in all situations and the styling happens with `disabled` class.

I feel this could probably be improved on. Maybe having something like:
```
this.paymentConfirmationButton = `${this.paymentStepSection} #payment-confirmation button`;
this.paymentConfirmationDisabledButton = `${this.paymentStepSection} #payment-confirmation button:not([disabled])`;
```
I had to double check the source and code a few times before I trusted that `this.paymentConfirmationButton` selector :)


Also if you could give input on the `${this.paymentStepSection} div.content > p` I feel that the `p` should at least have a class (something like "NoPaymentNeededParagraph") for easier styling and even just for this test to have more specific selector for the element.

Reply:
> `this.noPaymentNeededElement` is used in the test case.
> 
> What I meant with my comment is that the `this.paymentConfirmationButton` has the `[disabled]` selector. This might be confusing as the one might want to check that the button is there and what its state is. Also it looks like the disabled attribute is not even set in all situations and the styling happens with `disabled` class.
> 
> I feel this could probably be improved on. Maybe having something like:
> 
> ```
> this.paymentConfirmationButton = `${this.paymentStepSection} #payment-confirmation button`;
> this.paymentConfirmationDisabledButton = `${this.paymentStepSection} #payment-confirmation button:not([disabled])`;
> ```

> I had to double check the source and code a few times before I trusted that `this.paymentConfirmationButton` selector :)

You are absolutely right, your comment is the best way to do it. The main reason why we didn't have 2 selectors, is we didn't have enough use cases to use them both, but with this, I'm sure we can.

> Also if you could give input on the `${this.paymentStepSection} div.content > p` I feel that the `p` should at least have a class (something like "NoPaymentNeededParagraph") for easier styling and even just for this test to have more specific selector for the element.

I'm not sure where to do it. I will try to find and do it later